flow/project.py

```diff
@@ -933,19 +1091,34 @@ def _create_run_job_operations(self, entrypoint, default_directives, job,
         :rtype:
             Iterator[JobOperation]
         """
+
         for name, op in self.operations.items():
-            if op.eligible(job, ignore_conditions):
-                directives = self._resolve_directives(name, default_directives, job)
-                uneval_cmd = functools.partial(self._run_cmd, entrypoint=entrypoint,
-                                               operation_name=name, operation=op,
-                                               directives=directives, job=job)
-                # Uses a different id than the groups direct id. Do not use this for submitting
-                # jobs as current implementation prevents checking for resubmission in this case.
-                # The different ids allow for checking whether JobOperations created to run directly
-                # are different.
-                job_op = JobOperation(self._generate_id(job, name, index=index), name, job,
-                                      cmd=uneval_cmd, directives=deepcopy(directives))
-                yield job_op
+            filter = self.flow_select[name]
+            grouper, sort = self.flow_aggregate[name]
+            jobs_list = filter(jobs)
+            if sort is not None:
+                jobs_list = sort(jobs_list)
+            jobs_list = grouper([job for job in jobs_list])
+            jobs_list = [[j for j in job] for job in jobs_list]
```

Comment:
From my understanding, the first line returns a generator over groups of jobs correct? if so why do we need to bother explicitly making it a list here? Also, grouper should work if passed the filter object correct?

Reply:
We're just ensuring list here. To optimize this, I can ensure the list inside the loop rather than creating a list as a whole.